metrics/opencensus_exporter.go

```diff
@@ -14,19 +14,38 @@ limitations under the License.
 package metrics
 
 import (
+	"context"
+	"crypto/tls"
+	"fmt"
+
+	"go.opencensus.io/resource"
+	"go.opencensus.io/tag"
+	"knative.dev/pkg/metrics/metricskey"
+
 	"contrib.go.opencensus.io/exporter/ocagent"
 	"go.opencensus.io/stats/view"
 	"go.uber.org/zap"
+	"google.golang.org/grpc/credentials"
+	"k8s.io/apimachinery/pkg/api/errors"
+	clientv1 "k8s.io/client-go/listers/core/v1"
+	"knative.dev/pkg/system"
 )
 
 func newOpenCensusExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
```

Comment:
Separate from this PR, but it looks like WithServiceName does not have the same effects as component name does for Prometheus/Stackdriver backends.

For example, for Prometheus I'm seeing:
"namespace_actual_pods" instead of "autoscaler_actual_pods"
And Stackdriver I get:
"custom.googleapis.com/opencensus/actual_pods" instead of "custom.googleapis.com/opencensus/autoscaler/actual_pods"

For Stackdriver, it looks like setting the "view.Name" field for each view in stats_reporter to "autoscaler/actual_pods" would work. Haven't looked into Prometheus yet.

Reply:
Thanks -- do you have quick instructions for setting up the environment you're using to validate this?

Reply:
I filled out the Stackdriver section of the config from your other PR, then deployed on GKE cluster, then the metrics show up in Stackdriver under 
"custom.googleapis.com/opencensus/actual_pods" with Monitored Resource Type "global".

The "testing" prefix only seems to show up in the display name, but not the metric name itself.

```
apiVersion: v1
kind: ConfigMap
metadata:
  labels:
    serving.knative.dev/release: devel
  name: opencensus-agent-config
  namespace: knative-monitoring
data:
  opencensus-agent-config: |
    receivers:
      opencensus:
        address: ":55678"
    #   jaeger: {}
    #   zipkin: {}
    exporters:
      stackdriver:
        # project: "kubecon-knative-2019"  # Optional, defaults to node project
        metric_prefix: "testing"
        enable_metrics: true
        enable_tracing: true
      prometheus:
        namespace: "namespace"
        address: ":8888"
# ---
# apiVersion: v1
# kind: Secret
# metadata:
#   name: stackdriver-key
#   namespace: knative-monitoring
# type: Opaque
# data:
#   credentials.json: <<<BASE64 JSON>>>
---
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  labels:
    serving.knative.dev/release: devel
  name: opencensus-agent
  namespace: knative-monitoring
spec:
  selector:
    matchLabels:
      component: opencensus-agent
  template:
    metadata:
      labels:
        component: opencensus-agent
        serving.knative.dev/release: devel
    spec:
      containers:
      - command:
        - /ocagent_linux
        - --config=/conf/opencensus-agent-config.yaml
        # env:
        # - name: GOOGLE_APPLICATION_CREDENTIALS
        #   value: /secrets/credentials.json
        image: omnition/opencensus-agent:0.1.6
        name: opencensus-agent
        ports:
        - containerPort: 55678
          name: collect
          protocol: TCP
        - containerPort: 55679
          name: zpages
          protocol: TCP
        - containerPort: 8888
          name: prometheus
          protocol: TCP
        resources:
          limits:
            cpu: 500m
            memory: 500Mi
          requests:
            cpu: 100m
            memory: 100Mi
        volumeMounts:
        - mountPath: /conf
          name: opencensus-agent-config-vol
        # - mountPath: /secrets
        #   name: stackdriver-key
      volumes:
      - configMap:
          items:
          - key: opencensus-agent-config
            path: opencensus-agent-config.yaml
          name: opencensus-agent-config
        name: opencensus-agent-config-vol
      # - name: stackdriver-key
      #   secret:
      #     defaultMode: 420
      #     secretName: stackdriver-key
---
apiVersion: v1
kind: Service
metadata:
  labels:
    serving.knative.dev/release: devel
  name: opencensus
  namespace: knative-monitoring
spec:
  ports:
  - name: opencensus
    port: 55678
    protocol: TCP
    targetPort: 55678
  selector:
    component: opencensus-agent
  type: ClusterIP
```